### PR TITLE
[compiler] Fixture to show current behavior of nested function declarations

### PR DESCRIPTION
**Pull request by mvitousek**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #30922
* __->__ #30921

Summary:
Shows both memoized and unmemoized function declarations

### CODE REVIEW

File: compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/function-decl-nested.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  function g(x) {
+    return 2;
+  }
+  g();
+  function f() {
+    return x.x;
+  }
+  function h(x) {
+    return 2;
+  }
```
Comment (josephsavona) [inline compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/function-decl-nested.js:13]: are we intentionally not using x? maybe add a comment with why not

**Review by josephsavona — APPROVED** [review]
Ok...the example isn't nested though? I don't quite follow what this fixture is trying to test

**Comment by mvitousek** [comment]
The idea here was to document the changes in the way we emit function declarations after the next diff... which in retrospect, I muddled by modifying the example in that diff to include additional function calls :|

**Comment by mvitousek** [comment]
Actually not really a useful test case at all, looking more at it

**Comment by Mishap111** [comment]
> The idea here was to document the changes in the way we emit function declarations after the next diff... which in retrospect, I muddled by modifying the example in that diff to include additional function calls :| 

dimension split [L²]___dim¹/area³

**Pull request closed without merging**